Server/ReasnAPI/ReasnAPI/Services/EventService.cs

```diff
@@ -1,16 +1,291 @@
+
+            var existingParamsInDb = context.Parameters
+                .Where(param => newParameters.Any(newParam => newParam.Key == param.Key && newParam.Value == param.Value))
+                .ToList();
+
+            var paramsToAdd = newParameters
+                .Where(newParam => existingParamsInDb.All(existingParam => existingParam.Key != newParam.Key || existingParam.Value != newParam.Value))
+                .ToList();
+
+            context.Parameters.AddRange(paramsToAdd);
+            context.SaveChanges();
+
+            var paramsToAttach = newParameters
+                .Where(newParam => eventToUpdate.Parameters.All(existingParam => existingParam.Key != newParam.Key || existingParam.Value != newParam.Value))
+                .ToList();
+
+            paramsToAttach.ForEach(eventToUpdate.Parameters.Add);
+
+            context.SaveChanges();
+        }
+    }
+
+    public void DeleteEvent(int eventId)
+    {
+        using (var scope = new TransactionScope())
+        {
+            var eventToDelete = context.Events.Include(e => e.Tags).Include(e => e.Parameters).Include(e => e.Participants).Include(e => e.Comments).FirstOrDefault(e => e.Id == eventId);
+
+            if (eventToDelete is null)
+            {
+                throw new NotFoundException("Event not found");
+            }
+
+            RemoveTags(eventToDelete, eventId);
+            RemoveParameters(eventToDelete, eventId);
+            RemoveComments(eventToDelete);
+            RemoveParticipants(eventToDelete);
+
+            context.Events.Remove(eventToDelete);
+            context.SaveChanges();
+            scope.Complete();
+        }
+    }
+
+    private void RemoveTags(Event eventToDelete, int eventId)
+    {
+        foreach (var tag in eventToDelete.Tags.ToList())
+        {
+            eventToDelete.Tags.Remove(tag); // remove the association first
+            // Check if the tag is associated with any other event
+            if (!context.Events.Any(e => e.Tags.Any(t => t.Name == tag.Name) && e.Id != eventId) && !context.Events.Any(e => e.Tags.Contains(tag)))
+            {
+                context.Tags.Remove(tag); // then remove the tag
+            }
+        }
+    }
+
+    private void RemoveParameters(Event eventToDelete, int eventId)
+    {
+        foreach (var parameter in eventToDelete.Parameters.ToList())
+        {
+            eventToDelete.Parameters.Remove(parameter); // remove the association first
+            // Check if the parameter is associated with any other event
+            if (!context.Events.Any(e => e.Parameters.Any(p => p.Key == parameter.Key && p.Value == parameter.Value) && e.Id != eventId) && !context.Events.Any(e => e.Parameters.Contains(parameter)))
+            {
+                context.Parameters.Remove(parameter); // then remove the parameter
+            }
+        }
+    }
+
+    private void RemoveComments(Event eventToDelete)

```

Comment:
Can't we just make parameters of collection of comments? The name is misleading 'remove comments' but which comments if we don't pass any?